### ISSUE TITLE
Painfully slow "Validating GWT components"

### ISSUE THREAD

**Issue by slavap** [gwt-plugins/gwt-eclipse-plugin#370]
Not sure that GWT plugin is responsible for that.
On project clean validation starts, and it takes more than a minute per validation of each hundred java/ui.xml files in my project on very high end desktop PC. If plugin is responsible for that, then may be it's good idea to check MD5 hash of files before validating them (may be they are not changed since previous validation).

**Comment by slavap** [comment]
@branflake2267 No, it’s not new, but getting worse with my project size increased.

**Comment by branflake2267** [comment]
Okay, I know what you're talking about. I've noticed this issue too. Thanks for reporting. I'll mark it as a bug and take a look at it on the next go.

**Comment by CrushaKRool** [comment]
Could this be related to (and maybe fixed by) #383?

**Comment by slavap** [comment]
@CrushaKRool @branflake2267 I've installed new version of GWT plugin from http://storage.googleapis.com/gwt-eclipse-plugin/v3/snapshot, and unfortunately no performance improvement for this issue :-(

**Comment by branflake2267** [comment]
Bummer. I'd like to help fix it too! I'm about a week or two weeks off before I can spend more time. @slavap can you tell me more about your project config?

**Comment by slavap** [comment]
@branflake2267 Nothing special about my project config. Typical Maven project with GWT facet. I think the problem is in enormous number of ui.xml files, I have around 400 of them. And during "Validating GWT components" I see that all java files with @UiTemplate annotations are processed. So if I understand it properly, GWT is making sure that all @UiField have corresponding declarations in ui.xml

**Comment by slavap** [comment]
@branflake2267 I've updated my Eclipse STS to the latest version 4.10 today, and I don't see "Validating GWT components" step on project clean anymore :-) Everything else is working just fine, so I'm really happy how it is right now :-) Don't know if Eclipse fixed something, or there is some incompatibility with GWT plugin, but clean is fast now and I don't see any errors in log.

**Comment by protoism** [comment]
@slavap, I really think that this issue might not be relevant anymore.
Can you test on main branch (https://eclipse.gwtproject.org.vertispan.com/2023-01-18/ should be ok) if the problem is gone?
I really think that it was more related to STS and, if I remember it correctly, to gradle integration

**Comment by slavap** [comment]
@protoism still present - though with fast SSD and processor it is not so painful. What I see that plug-in goes through all ui.xml files after each “project clear” or “maven update”, May be also some other activity.

**Comment by protoism** [comment]
@slavap Honestly speaking, I doubt that the issue can be easily solved. All changes being made are about making the plugin just build without regressions... touching the plugin... who knows if and when we will be able.
Moreover, uibinder stuff is not that popular nowadays!

Anyway, if you want to spend a few minutes on this, I'd say that this information might be useful:

* Verify that there is no O/S interference, say... antivirus (sorry for the doubt)
* Profile eclipse and try to get a plausible stack trace of hotspots (https://visualvm.github.io/ should be ok)
* Why not, a small movie showing the problem

**Comment by slavap** [comment]
@protoism 
>> All changes being made are about making the plugin just build without regressions... touching the plugin... who knows if and when we will be able.
I'm very glad that this step was actually done. No high expectations about anything else.

1. Sure thing, no antivirus active.
2. Profile eclipse and try to get a plausible stack trace of hotspots << not sure I'm qualified enough for such things :-)
3. Did additional testing - and I think we can close this issue for now. On my huge project "Validating GWT components" currently takes about 40 seconds (on nvme SSD and 4+GHz CPU) - and I think that is kind of acceptable time.
